docs/contributor/arch/01-collector-structure.md

```diff
@@ -0,0 +1,32 @@
+# 1. Collector Structure
+
+Date: 2024-11-20
+
+## Status
+
+Proposed
+
+## Context
+
+The current implementation of KMCs SKR processing has been built with a monolithic architecture. This has led to a number of issues including:
+- a single issue processing an SKR renders the entire billing process for this cluster invalid
+- adding new ressources to the billing process is difficult
+- the current implementation is hard to test
+- every client needs to implement 
+
+## Decision
+
+We will refactor the current implementation of the SKR processing to a more modular architecture. This will allow us to:
+- process each scan result independently
+- add new resources to scan easily
+- each scan result will contain its own logic for processing the data. Processing will include:
```

Comment:
`This will allow us to each scan will contain its own logic...`
This sentence doesn't make sense.

Neither do the following sentences.
`This will allow us to extracting all requited information...`
`This will allow us to converting the scan result`

How about:
line 19: "This has the following advantages:
- We can process...
- We can add...
- Each new scan result will...
- We can extract... {if it's "we" who do that?}
- We can convert... {if it's "we" who do that?}"